src/handlers/opportunitiesHandlers.ts

```diff
@@ -33,6 +38,50 @@ const getOpportunities = (req: Request, res: Response): void => {
     });
 };
 
+const getOpportunity = (req: Request, res: Response): void => {
+  const isInteger = (it: string): boolean => !Number.isNaN(parseInt(it, 10))
+    && parseInt(it, 10) % 1 === 0;
+
+  if (!isInteger(req.params.id)) {
+    res.status(400)
+      .contentType('application/json')
+      .send({ message: 'Bad request. Send an integer id or call with no parameters.' });
+    return;
+  }
+
+  logger.debug(req);
```

Comment:
I think by default it is logging the fulfilled request, in other words the request and response after it was served. Perhaps that would include enough detail in the error to see what we are trying to see here, perhaps not. The placement of the statement after validation of `req.params.id` is related to its first real use on the following line 53, where it uses the id in the query. If id isn't an integer, it might be unsafe to print the request at all, so that is probably why I had it on the next line. Regardless, I am OK with removing this one and re-adding it locally as needed, or demoting it to trace.